src/tree/Element.mjs

```diff
@@ -623,10 +623,18 @@ export default class Element {
         }
 
         if (sourceChanged) {
+            const texture = this.__displayedTexture;
             if (this.__displayedTexture) {
-                this.emit('txLoaded', this.__displayedTexture);
+                // Emitting directly causes recursion limit when
+                // texture source is accessed from within its callback.
+                // https://github.com/rdkcentral/Lightning/issues/447
+                setTimeout(() => {
+                    this.emit('txLoaded', texture);
```

Comment:
Perhaps `removeUpdateSourceTexture()` can just be called prior to the emission of 'txLoaded'?

Reply:
@frank-weindel The update would indeed skip to the next available frame and while it wouldn't affect performance it might be an undesired side effect. Your suggestion looks like a good alternative.